city_scrapers/spiders/il_procurement_policy.py

```diff
@@ -0,0 +1,153 @@
+        """Parse or generate links."""
+        links = []
+        title_str = " ".join(item.css("*::text").extract()).strip()
+        if 'pdf' in title_str:
+            title_str = re.sub("Agenda.pdf", "", title_str).strip()
+            title_str += " Agenda"
+        links.append({
+            'title': title_str,
+            'href': response.urljoin(item.attrib['href']),
+        })
+        return links
+
+    def _parse_past_links(self, response):
+        """parse start datetime as a naive datetime object"""
+        links = []
+        index = None
+        for item in response.css(".ms-rtestate-field p a"):
+            title_str = " ".join(item.css("*::text").extract()).strip()
+            title_str = title_str.replace("\u200b", "")
+            if '.pdf' in title_str:
+                index = title_str.index('.pdf')
+            if '[' in title_str:
+                index = title_str.index('[')
+            if '-' in title_str:
+                index = title_str.index('-')
+            title_str = title_str[:index]
+            links.append({'title': title_str, 'href': response.urljoin(item.attrib['href'])})
+        for item in response.css(".ms-rtestate-field .list-unstyled li a"):
+            title_str = " ".join(item.css("*::text").extract()).strip()
+            title_str = re.sub(".pdf", "", title_str).strip()
+            title_str = title_str.replace("\u200b", "")
+            index = len(title_str)
+            if '- Amended' in title_str:
+                title_str = title_str.replace("- Amended", "").strip()
+            if '[' in title_str:
+                index = title_str.index('[')
+            if '-' in title_str:
+                index = title_str.index('-')
+            title_str = title_str[:index].strip()
+            links.append({'title': title_str, 'href': response.urljoin(item.attrib['href'])})
+        return links
+
+    def _past_start(self, item):
+        """parse or generate links from past meetings"""
+        time_object = time(10, 0)
+        date_str = item['title']
+        if not len(date_str):
+            date_str = "December 2, 2015"
+            date_object = datetime.strptime(date_str, "%B %d, %Y").date()
+            return datetime.combine(date_object, time_object)
+        date_object = datetime.strptime(date_str, "%B %d, %Y").date()
+        return datetime.combine(date_object, time_object)
+
+    def _parse_source(self, response):
+        """Parse or generate source."""
+        return response.url
+
+    def _upcoming_meetings(self, response):
+        for item in response.css(".ms-rtestate-field p strong a"):
+            meeting = Meeting(
+                title='Procurement Policy Board',
+                description='',
+                classification=self._parse_classification(item),
+                start=self._parse_start(item),
+                end=None,
+                all_day=False,
+                time_notes='End time not specified',
```

Comment:
Leaving out the end time is the default, so we can leave this blank

Reply:
`time_notes` should still be an empty string

Reply:
So for the 'end' I should just take it out of the .py file and the test file I assume?